### ISSUE TITLE
Автозагрузка модуля с определенными параметрами перестала работать в версии v1.0.0-rc5

### ISSUE THREAD

**Issue by alekskomp** [Waujito/youtubeUnblock#206]
### Bug description

Параметры указанные в 

- /etc/modprobe.d/kyoutubeUnblock.conf
```
options kyoutubeUnblock verbosity=silent exclude_domains=ggpht.com
```

перестали работать в версии v1.0.0-rc5

В логе:

```
kernel: kyoutubeUnblock: unknown parameter 'verbosity' ignored
kernel: kyoutubeUnblock: unknown parameter 'exclude_domains' ignored
```

Получилось изменить только через 
` echo "--silent --exclude-domains=ggpht.com" | tee /sys/module/kyoutubeUnblock/parameters/parameters`

Изменился синтаксис? Как теперь задать параметры при старте модуля?

### Linux distribution

Debian 12.7

### Device architecture

x86_64
11th Gen Intel(R) Core(TM) i5-11400 @ 2.60GHz

### Configuration

--silent --exclude-domains=ggpht.com

**Comment by Ser9ei** [comment]
@Waujito удалось запустить новую версию сервиса после изменений в скрипте /etc/init.d/youtubeUnblock
Вероятно в коде скрипта остались legacy параметры (или имена параметров не соотвествуют коду в bin) для заупска /usr/bin/youtubeUnblock
Ниже см. пример ошибки из лога.

Пример  workaround в скрипте для запуска сервиса (# - не рабочий код, за ним следующая строка с внесенным изменением кода):
```
#	config_get_bool value "$config" tls_enabled 0
#	if [ $value -gt 0 ]; then
#		xappend "tls" "enabled"
#	else
#		xappend "tls" "disabled"
#	fi

#	append_opts "$config" fake_sni_seq_len fake_sni_type fake_custom_payload faking_strategy faking_ttl fake_seq_offset frag frag_sni_pos fk_winsize seg2delay synfake_len sni_detection udp_mode udp_fake_seq_len udp_fake_len udp_filter_quic
	append_opts "$config" fake_sni_seq_len fake_sni_type fake_custom_payload faking_strategy faking_ttl fake_seq_offset frag frag_sni_pos fk_winsize seg2delay synfake_len sni_detection
```

Пример ошибки в логе:

> daemon.err youtubeUnblock[29164]: /usr/bin/youtubeUnblock: unrecognized option: tls=enabled
> daemon.info youtubeUnblock[29164]: youtubeUnblock 1.0.0-5
> daemon.info youtubeUnblock[29164]: Bypasses deep packet inspection systems that rely on SNI
> daemon.info youtubeUnblock[29164]:
> daemon.info youtubeUnblock[29164]: Usage: /usr/bin/youtubeUnblock [ OPTIONS ]
> daemon.info youtubeUnblock[29164]: Options:
> daemon.info youtubeUnblock[29164]: 	--queue-num=<number of netfilter queue>
> daemon.info youtubeUnblock[29164]: 	--sni-domains=<comma separated domain list>|all
> daemon.info youtubeUnblock[29164]: 	--exclude-domains=<comma separated domain list>
> daemon.info youtubeUnblock[29164]: 	--fake-sni={1|0}
> daemon.info youtubeUnblock[29164]: 	--fake-sni-seq-len=<length>
> daemon.info youtubeUnblock[29164]: 	--fake-sni-type={default|random|custom}
> daemon.info youtubeUnblock[29164]: 	--fake-custom-payload=<hex payload>
> daemon.info youtubeUnblock[29164]: 	--fake-seq-offset=<offset>
> daemon.info youtubeUnblock[29164]: 	--faking-ttl=<ttl>
> daemon.info youtubeUnblock[29164]: 	--faking-strategy={randseq|ttl|tcp_check|pastseq|md5sum}
> daemon.info youtubeUnblock[29164]: 	--synfake={1|0}
> daemon.info youtubeUnblock[29164]: 	--synfake-len=<len>
> daemon.info youtubeUnblock[29164]: 	--frag={tcp,ip,none}
> daemon.info youtubeUnblock[29164]: 	--frag-sni-reverse={0|1}
> daemon.info youtubeUnblock[29164]: 	--frag-sni-faked={0|1}
> daemon.info youtubeUnblock[29164]: 	--frag-middle-sni={0|1}
> daemon.info youtubeUnblock[29164]: 	--frag-sni-pos=<pos>
> daemon.info youtubeUnblock[29164]: 	--fk-winsize=<winsize>
> daemon.info youtubeUnblock[29164]: 	--quic-drop
> daemon.info youtubeUnblock[29164]: 	--sni-detection={parse|brute}
> daemon.info youtubeUnblock[29164]: 	--seg2delay=<delay>
> daemon.info youtubeUnblock[29164]: 	--threads=<threads number>
> daemon.info youtubeUnblock[29164]: 	--packet-mark=<mark>
> daemon.info youtubeUnblock[29164]: 	--silent
> daemon.info youtubeUnblock[29164]: 	--trace
> daemon.info youtubeUnblock[29164]: 	--no-gso
> daemon.info youtubeUnblock[29164]: 	--no-ipv6
> daemon.info youtubeUnblock[29164]: 	--daemonize
> daemon.info youtubeUnblock[29164]: 	--noclose
> daemon.info youtubeUnblock[29164]: 	--syslog
> daemon.info youtubeUnblock[29164]: 	--fbegin
> daemon.info youtubeUnblock[29164]: 	--fend
> daemon.info youtubeUnblock[29164]:
> daemon.info youtubeUnblock[29164]: Unable to parse args: No error information
>

**Comment by Waujito** [comment]
@Ser9ei предположу, что вы самостоятельно собирали всё это дело для openwrt? Тогда скорее всего это я просто забыл хеш коммита в ветке поменять)

Всё исправил.

**Comment by Waujito** [comment]
> Изменился синтаксис?

Да, теперь всё задаётся через `/sys/module/kyoutubeUnblock/parameters/parameters`, параметры такие же, как в обычном анблоке.

Я очень доволен этим изменением, прописывать каждый раз новый параметр отдельно для модуля было максимально неприятно.

А вот кстати над автозагрузкой надо еще подумать

**Comment by Ser9ei** [comment]
@Waujito  да всё так, спасибо!

**Comment by alekskomp** [comment]
> > Изменился синтаксис?
> 
> Да, теперь всё задаётся через `/sys/module/kyoutubeUnblock/parameters/parameters`, параметры такие же, как в обычном анблоке.
> 
> Я очень доволен этим изменением, прописывать каждый раз новый параметр отдельно для модуля было максимально неприятно.
> 
> А вот кстати над автозагрузкой надо еще подумать


Т.е. сейчас через файл `/etc/modprobe.d/kyoutubeUnblock.conf` вообще нет возможности задать параметры модуля?

**Comment by Ser9ei** [comment]
Если никто не против, продолжу вопросы по новой версии здесь же.

Проверил сервис на роутере с mips архитектурой, OpenWrt 21 (соотвественно fw3 - iptables).
Установил по инстуркции (возможно её стоит актуализировать, например, в части замечания про lua для luci)

1. Новая версия luci-app на jsp
Ошибка при открытии на закладке "Serivce Status" (Configuration ошибок не выдавала, но и особо в ней не работал)

> 
> NetworkError
> 
> HTTP error 404 while loading class file "/luci-static/resources/tools/views.js?v=git-23.093.57360-e98243e"
>   at compileClass (https://192.168.1.1/luci-static/resources/luci.js?v=git-23.093.57360-e98243e:168:16)

Я так понимаю, это из за новых фич для визуализации статуса?

2. Пакет "youtubeUnblock" запускается, ошибок не выдает в логе, но и YT "не ускоряется".
Не разбирался, возможно нужны "другие" настройки, отличные от тех, что по умолчанию (на OpenWrt 23 с nft проблем не было, работает с настройками по умолчанию).
Если дадите наводку, какие параметры смотреть для совместимости с iptables - буду благодарен

3. "Ядерный" пакет заработал.
Пришлось выключить offloading, с активным - не работает.
Если шанс запускать с offloading, проблема с этим cpu только в производительности, без offloading 100Мбит тариф выдает только ~40-60Мбит/с

**Comment by Waujito** [comment]
> * Пакет "youtubeUnblock" запускается, ошибок не выдает в логе, но и YT "не ускоряется".
>   Не разбирался, возможно нужны "другие" настройки, отличные от тех, что по умолчанию (на OpenWrt 23 с nft проблем не было, работает с настройками по умолчанию).
>   Если дадите наводку, какие параметры смотреть для совместимости с iptables - буду благодарен
> 
> * "Ядерный" пакет заработал.
>   Пришлось выключить offloading, с активным - не работает.
>   Есть ли здесь шанс запускать с offloading?, проблема с этим cpu только в производительности, без offloading 100Мбит тариф выдает только ~40-60Мбит/с

offloading также влияет и на обычный анблок. Посмотрите #199

**Comment by Ser9ei** [comment]
> Посмотрите #199
Да, я могу тоже заметить, на этом же "однопотоковом" mips роутере OpenWrt 23 работал без отключения offloading, как писали по ссылке. С 21 не прошло, видимо на 21й нужна задержка)

**Comment by Waujito** [comment]
> Да, я могу тоже заметить, на этом же "однопотоковом" mips роутере OpenWrt 23 работал без отключения offloading, как писали по ссылке. С 21 не прошло, видимо на 21й нужна задержка)

А, так в iptables вроде бы нельзя контроллировать offloading

**Comment by Ser9ei** [comment]
И насколько заметил, в OpenWrt 23 на mt7620 не работает offloading.
Разницы между вкл/выкл offloading не заметил, возможно эта и причина почему ‘ускорение’ заработала на этой процессоре сразу же, без выключения offloading, в отличии от OpenWrt 21. Но жто мой опыт и эекспкремент не продолжил, с этим процессором на свежих релизах OpenWrt все печально с производительностью
